go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1042,6 +1042,12 @@ func (bnc *BaseNetworkController) setupGressPolicy(np *networkPolicy, gp *gressP
 		gp.addIPBlock(peer.IPBlock)
 		return nil, nil
 	}
+
+	if !bnc.doesNetworkRequireIPAM() {
```

Comment:
Maybe checking this earlier, e.g. in `convertMultiNetPolicyToNetPolicy` would be better

Reply:
It's going to be hard to do that inside that function, since that has no clue about the `net-attach-def` object, which makes it hard for us to understand if a network has IPAM or not. 


I'll try to think of something though ... I can definitely fail sooner than what I'm doing now.

Reply:
I'll explore these options.

IMO the first one is the cleaner option.